### PR TITLE
Version 5.0.0

### PR DESCRIPTION
**Pull request by FabienLavocat**
- **Breaking:** Remove Communications APIs.
- **Breaking:** Simplify the type exports for Millicast and Media APIs.
- Millicast - Add Cloud Transcoder APIs (beta).
- Millicast - Add more APIs (Webhook, Record Files...).
- Millicast - Fix stream APIs.
- Millicast - Fix publish token disable API.
- Update GitHub Actions.
- Add [js-logger](https://github.com/jonnyreeves/js-logger) as dependency and expose `Logger` object.

### CODE REVIEW

File: src/streaming/transcoders.ts [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+        isDescending: (options.isDescending ?? false).toString(),
+    };
+
+    if (options.status) {
+        params['status'] = options.status;
+    }
+
+    const queryOptions = {
+        hostname: Urls.getRtsHostname(),
+        path: '/api/transcoders',
```
Comment (j12y) [inline src/streaming/transcoders.ts:41]: Noting that this is correct, it isn't like some of the other endpoints that are of the rpc style form `/api/resource/list`

File: src/streaming/transcoders.ts [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+ * @remarks Cloud transcoder usage is not currently available for general usage.
+ * If you would like to opt in, please contact our Sales team.
+ *
+ * @param apiSecret The API Secret used to authenticate this request.
+ * @param publishToken Information about the new publish token.
+ *
+ * @returns A {@link !Promise Promise} whose fulfillment handler receives a {@link Transcoder} object.
+ */
+export const createTranscoder = async (apiSecret: string, transcoder: CreateTranscoderRequest): Promise<Transcoder> => {
+    if (transcoder.startNow == null) {
```
Comment (j12y) [inline src/streaming/transcoders.ts:118]: `startNow` and `passThrough` are optional parameters, but is **false** the default for `startNow`?

**Review by j12y — APPROVED** [review]
Visual inspection of code, things look good. I do always appreciate your attention to writing clean code.

I could still spend more time testing, but I don't want that to hold up you moving forward because overall and the nature of the changes it seems fine.

File: src/index.ts [hunk 3]
```diff
@@ -1,27 +1,21 @@
...
  */
 export * as streaming from './streaming';
 
 /** @ignore */
 export * as urls from './urls';
 
+// Expose the logger to control when/where to log
+export { default as Logger } from 'js-logger';
+
 declare const __VERSION__: string;
```
[the remark this answers is not in the document]
Comment (FabienLavocat) [reply src/index.ts:19]: Correct, too many breaking changes, so I bumped the version to 5.0.0.

File: src/streaming/transcoders.ts (hunk 2 above)
Comment (FabienLavocat) [reply src/streaming/transcoders.ts:118]: It should be `false`, I'm changing it.

**Review by j12y — APPROVED** [review]
Missed that you updated this, i like the changes to the readme.

See that startNow is also consistent.

👍

**Pull request merged**